### PR TITLE
Spanner add back partitioned DML support

### LINKED ISSUE #1709: Add option to let users execute partitioned DML in Spanner

### PR DESCRIPTION
**Pull request by ChengyuanZhao**
fixes #1709 
adds partitioned DML back to spanner

### CODE REVIEW

**Comment by ChengyuanZhao** [comment]
uh oh, let me do some rebase...

**Comment by ChengyuanZhao** [comment]
@dmitry-s @meltsufin merged, ready for review PTAL

File: spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java [hunk 1]
```diff
@@ -152,6 +153,17 @@ public long executeDmlStatement(Statement statement) {
 		return rowsAffected;
 	}
 
+	@Override
+	public long executePartitionedDmlStatement(Statement statement) {
+		Assert.notNull(statement, "A non-null statement is required.");
+		maybeEmitEvent(new BeforeExecuteDmlEvent(statement));
+		long rowsAffected = doWithOrWithoutTransactionContext(x -> {
+			throw new SpannerDataException("Cannot execute partitioned DML in a transaction.");
+		}, () -> this.databaseClientProvider.get().executePartitionedUpdate(statement));
```
Comment (dmitry-s) [inline spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java]: is it properly formatted?

Comment (ChengyuanZhao) [reply spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java]: yes I did this manually and then did auto reformat which did not change this at all. alas, the checkstyle passes.

**Pull request merged**